library/src/precond/bisection_algorithm.h

```diff
@@ -0,0 +1,132 @@
+auto bisection_algorithm = [=](auto const row) -> void {
```

Comment:
You need to insert the copyright header. You'll find it in any other files

Reply:
I don't think it would make any difference if you create a proper inlined device function rather than a lambda one.

Reply:
use \#pragma once and the ifndef/define macros